backend/apps/matching/src/matching.service.ts

```diff
@@ -0,0 +1,104 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { ClientProxy } from '@nestjs/microservices';
+import { LookingToMatchModel } from './database/models/lookingToMatch.model';
+import { firstValueFrom } from 'rxjs';
+
+import { LookingToMatchDaoService } from './database/daos/lookingToMatch/lookingToMatch.dao.service';
+import { Service } from '@app/microservice/interservice-api/services';
+import { WebsocketServiceApi } from '@app/microservice/interservice-api/websocket';
+import { QuestionServiceApi } from '@app/microservice/interservice-api/question';
+import { CollaborationServiceApi } from '@app/microservice/interservice-api/collaboration';
+
+@Injectable()
+export class MatchingService {
+  constructor(
+    @Inject(Service.WEBSOCKET_SERVICE)
+    private readonly webSocketClient: ClientProxy,
+    @Inject(Service.QUESTION_SERVICE)
+    private readonly questionClient: ClientProxy,
+    @Inject(Service.COLLABORATION_SERVICE)
+    private readonly collaborationClient: ClientProxy,
+    private readonly lookingToMatchDaoService: LookingToMatchDaoService,
+  ) {}
+
+  async findMatch(matchingEntry: Partial<LookingToMatchModel>) {
+    this.lookingToMatchDaoService.updateMatchingEntryIfExist({
+      ...matchingEntry,
+      isConnected: true,
+    });
```

Comment:
We don't need this if our websocket state is consistently with the entries in the db? Also can remove `isConnected`. To make it consistent, we can implement the `handleDisconnect` method in the websocket.gateway

Reply:
It could change is say users start a new match without disconnecting